### PR TITLE
Remove TrieStore allocations by using ArrayPoolList

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes:
- Remove TrieStore allocations by using ArrayPoolList
- Small refactors

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs [hunk 1]
```diff
@@ -412,36 +409,34 @@ private bool CanPruneCacheFurther()
             {
                 if (_logger.IsDebug) _logger.Debug("Elevated pruning starting");
 
-                List<BlockCommitSet> toAddBack = new();
-
-                List<BlockCommitSet> candidateSets = new();
-                while (_commitSetQueue.TryDequeue(out BlockCommitSet? frontSet))
+                using ArrayPoolList<BlockCommitSet> toAddBack = new(_commitSetQueue.Count * 2);
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:412]: Why *2?

**Review by asdacap — APPROVED** [review]

File: src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:412]: To have headroom if the queue grows during pruning

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs:412]: Also feel free to review ArrayPoolList code, though it is used in our code without issues already.

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs]: Ok I will remove it

**Pull request merged**